src/window.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
+export class WindowService {
+
+  get nativeWindow(): any {
+    return _window();
+  }
+}
+
+function _window(): any {
+  // return the global native browser window object
+  return typeof window !== 'undefined' ? window : undefined;
+}
```

Comment:
> which is the same as just using window directly

Are you sure? The main problem is that `node` will (under the server rendering) execute some of the code and since `window` is not defined in the node runtime it will throw an error (`ReferenceError: window is not defined`), right?
If we just wrap the reference in a function like `getNativeWindow` above it should work? Right?
<img width="745" alt="screen shot 2017-03-19 at 20 26 59" src="https://cloud.githubusercontent.com/assets/1525612/24084045/54b3e5fc-0ce3-11e7-8e9a-7538a4666704.png">